frontend/packages/data-portal/app/components/Deposition/RunRow.tsx

```diff
@@ -0,0 +1,245 @@
+                    length: Math.min(totalCount, MAX_PER_FULLY_OPEN_ACCORDION),
+                  },
+                  (_, index) => (
+                    <SkeletonAnnotationRow
+                      key={`annotation-skeleton-${index}`}
+                    />
+                  ),
+                )}
+              </>
+            )
+          }
+
+          // Show error state
+          if (error) {
+            return (
+              <TableRow className="border-b border-light-sds-color-semantic-base-divider">
+                <CellComponent colSpan={3}>
+                  <div className="pl-sds-xl py-sds-m text-sds-color-primitive-red-600">
+                    Error loading annotations
+                  </div>
+                </CellComponent>
+              </TableRow>
+            )
+          }
+
+          // Transform backend data to component format, use empty array when no data
+          const annotations =
+            data && 'annotationShapes' in data && data.annotationShapes
+              ? data.annotationShapes.map((shape) => ({
+                  id: shape.annotation?.id ?? shape.id,
+                  annotationName: shape.annotation?.objectName ?? '--',
+                  shapeType: shape.shapeType ?? '--',
+                  methodType: shape.annotation?.methodType ?? '--',
+                  depositedIn: '--',
+                  depositedLocation: '--',
+                  runName: run.runName,
+                  objectName: shape.annotation?.objectName,
+                  groundTruthStatus: shape.annotation?.groundTruthStatus,
+                  s3Path: shape.annotationFiles?.edges?.[0]?.node?.s3Path,
+                }))
+              : []
+
+          return (
+            <>
+              {annotations.map((annotation) => (
+                <TableRow
+                  key={`${run.runName}-${annotation.id}`}
+                  className="border-b border-light-sds-color-semantic-base-divider"
+                >
+                  <TableCell width={{ width: 350 }}>
+                    <div className="pl-sds-xl">
+                      <AnnotationNameTableCell
+                        annotationId={annotation.id}
+                        groundTruthStatus={annotation.groundTruthStatus}
+                        objectName={annotation.objectName}
+                        s3Path={annotation.s3Path}
+                      />
+                    </div>
+                  </TableCell>
+                  <TableCell width={{ width: 160 }}>
+                    <span className="text-sds-body-s-400-wide">
+                      {annotation.shapeType}
+                    </span>
+                  </TableCell>
+                  <MethodTypeCell
+                    methodType={
+                      annotation.methodType as Annotation_Method_Type_Enum
+                    }
+                  />
+                </TableRow>
+              ))}
+            </>
+          )
+        })()}
+    </Fragment>
+  )
+}
```

Comment:
This file's a bit long, but will be fine once the separate components are moved to their own files.